GetIntoTeachingApiTests/Controllers/OperationsControllerTests.cs

```diff
@@ -130,5 +131,56 @@ public void TriggerLocationSync_EnqueuesLocationSyncJob()
                                   (job.Method.Name == "RunAsync")),
                 It.IsAny<EnqueuedState>()));
         }
+
+        [Fact]
+        public void RemoveUnknownLocations_Authorize_IsPresent()
+        {
+            MethodInfo triggerLocationSync = typeof(OperationsController).GetMethod("RemoveUnknownLocations");
+            triggerLocationSync.Should().BeDecoratedWith<AuthorizeAttribute>();
+        }
+
+        [Fact]
+        public async Task RemoveUnknownLocations_WhenDryRunSpecified_ReturnsMessage()
+        {
+            const bool dryRun = true;
+            const int mockNumberOfLocations = 10;
+            var mockStore = new Mock<IStore>();
+            mockStore.Setup(store => store.GetNumberOfUnknownLocations())
+                .Returns(mockNumberOfLocations);
+            var controller = new OperationsController(

```

Comment:
All this setup is already done for you in the constructor, this test case could just be:

```
public async Task RemoveUnknownLocations_WhenDryRunSpecified_ReturnsMessage()
{
  _mockStore.Setup(store => store.GetNumberOfUnknownLocations())
                .Returns(10);

  var response = await controller.RemoveUnknownLocations(true);
  
  var response = response.Should().BeOfType<OkObjectResult>() as ObjectResult;
  response.Value.Should().Be("Number of locations with Unknown source: 10");
}
```

I would also add a `Verify` on the store to ensure the `RemoveUnknownLocations` is never called

Reply:
I did this in a separate test. I don't know how important "one thing per test" is when you write the tests after doing the thing. But, I figured in this case they were separate enough to warrant a new test?

Reply:
I would typically group assertions that are very closely related into the same test case, mainly to avoid repeated setup/teardown logic, but since that's not a big issue here I'm not fussed either way 👍